CRM/Mailing/BAO/Mailing.php

```diff
@@ -2637,22 +2637,8 @@ public function getReturnProperties() {
    */
   public static function commonCompose(&$form) {
     //get the tokens.
-    $tokens = CRM_Core_SelectValues::contactTokens();
+    $tokens = array();
 
-    $className = CRM_Utils_System::getClassName($form);
-    if ($className == 'CRM_Mailing_Form_Upload') {
-      $tokens = array_merge(CRM_Core_SelectValues::mailingTokens(), $tokens);
-    }
-    elseif ($className == 'CRM_Admin_Form_ScheduleReminders') {
-      $tokens = array_merge(CRM_Core_SelectValues::activityTokens(), $tokens);
-      $tokens = array_merge(CRM_Core_SelectValues::eventTokens(), $tokens);
-      $tokens = array_merge(CRM_Core_SelectValues::membershipTokens(), $tokens);
-    }
-    elseif ($className == 'CRM_Event_Form_ManageEvent_ScheduleReminders') {
-      $tokens = array_merge(CRM_Core_SelectValues::eventTokens(), $tokens);
-    }
-
```

Comment:
Tim - It looks like the old behavior was to expose ONLY eventTokens to ScheduleReminders form IF it is called from the Manage Events UI. Your refactor seems to expose activity, event and membership tokens for any instance of the ScheduleReminder form. Am I reading this correctly? Did you test what happens if use selects a membership token for an event-related reminder?